js/views/modals/listingDetail/Listing.js

```diff
@@ -307,6 +308,22 @@ export default class extends BaseModal {
     });
   }
 
+  startPurchase() {
+    const selectedVariants = [];
+    this.variantSelects.each((i, select) => {
+      const variant = {};
+      variant.name = $(select).attr('name');
+      variant.value = $(select).val();
+      selectedVariants.push(variant);
+    });
+
+    this.purchaseModal = launchPurchaseModal({
+      listing: this.model.toJSON(),
```

Comment:
I don't have enough context to give a strong opinion on this... but why are you passing in the flat JSON of the listing as opposed to the whole listing model? The listing model has some utility functions and I would think having access to its full API would be more efficient.

Am I missing something.